packages/marketplace/src/reducers/developer-subscriptions.ts

```diff
@@ -0,0 +1,93 @@
+import { Action } from '../types/core'
+import { isType } from '../utils/actions'
+import {
+  developerFetchSubscriptions,
+  developerFetchSubscriptionsSuccess,
+  developerCreateSubscription,
+  developerCreateSubscriptionSuccess,
+  developerCreateSubscriptionFalure,
+} from '@/actions/developer-subscriptions'
+import { PagedResultSubscriptionModel_, SubscriptionModel } from '@/services/developer-subscriptions'
+
+export type DeveloperSubscriptionsState = {
+  create: CreateDeveloperSubscriptionsState
+  list: ListDeveloperSubscriptionsState
+}
+
+export type CreateDeveloperSubscriptionsState = {
+  loading: boolean
+  subscription?: SubscriptionModel
+}
+
+export type ListDeveloperSubscriptionsState = {
+  data: PagedResultSubscriptionModel_ | null
+  loading: boolean
+}
+
+export const defaultState: DeveloperSubscriptionsState = {
+  create: {
+    loading: false,
+  },
+  list: {
+    loading: false,
+    data: null,
+  },
+}
+
+const developerSubscriptionsReducer = (
+  state: DeveloperSubscriptionsState = defaultState,
+  action: Action<any>,
+): DeveloperSubscriptionsState => {
+  if (isType(action, developerCreateSubscription)) {
+    return {
+      ...state,
+      create: {
+        loading: true,
+      },
+    }
+  }
+
+  if (isType(action, developerCreateSubscriptionSuccess)) {
+    return {
+      ...state,
+      create: {
+        loading: false,
+        subscription: action.data,
```

Comment:
No need to store the response after creating a subscription, it will make our redux tree more complex. Please use passing a callback function to sagas approach when dispatching actions instead